Goal: Add differentiated Makefile.inc files for optimized and non
      optimised compilation

Status wrt upstream: too Debian specific


Index: geneweb-4.10/tools/Makefile.inc.opt
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ geneweb-4.10/tools/Makefile.inc.opt	2005-08-30 08:13:40.000000000 +0200
@@ -0,0 +1,42 @@
+# $Id: 000_Makefile,v 1.2 2005/10/10 20:22:40 bubulle Exp $
+
+TOP=..
+CAMLP4_COMM=TOP=$(TOP) $(TOP)/tools/camlp4_comm.sh pa_ifdef.cmo
+CAMLP4F=-DUNIX
+CAMLP4D=`camlp4 -where`
+LIBUNIX=-cclib -lunix
+OCAMLC=ocamlc.opt -warn-error A
+OCAMLOPT=ocamlopt.opt -warn-error A
+OCAMLI=
+OCAMLD=
+STRIP=strip
+RM=/bin/rm -f
+EXE=
+STATIC=-ccopt -static
+STATIC=
+
+all::
+opt::
+
+clean::
+	$(RM) *.out *.opt *.cm[oixa] *.cmxa *.pp[oi] *.o *.a *.obj *.lib *.lck *.bak *~ .#*
+
+.SUFFIXES: .ml .cmo .mli .cmi .cmx
+
+.mli.cmi:
+	@if test `basename $<` != $<; then echo "Bad directory for $<"; exit 1; fi
+	@$(CAMLP4_COMM) $(CAMLP4F) $< -o $*.ppi
+	$(OCAMLC) $(OCAMLI) -I $(CAMLP4D) -c -intf $*.ppi
+	$(RM) $*.ppi
+
+.ml.cmo:
+	@if test `basename $<` != $<; then echo "Bad directory for $<"; exit 1; fi
+	@$(CAMLP4_COMM) $(CAMLP4F) $< -o $*.ppo
+	$(OCAMLC) $(OCAMLI) $(OCAMLD) -I $(CAMLP4D) -c -impl $*.ppo
+	$(RM) $*.ppo
+
+.ml.cmx:
+	@if test `basename $<` != $<; then echo "Bad directory for $<"; exit 1; fi
+	@$(CAMLP4_COMM) $(CAMLP4F) $< -o $*.ppo
+	$(OCAMLOPT) $(OCAMLI) -I $(CAMLP4D) -c -impl $*.ppo
+	$(RM) $*.ppo
Index: geneweb-4.10/tools/Makefile.inc.out
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ geneweb-4.10/tools/Makefile.inc.out	2005-08-30 08:13:40.000000000 +0200
@@ -0,0 +1,42 @@
+# $Id: 000_Makefile,v 1.2 2005/10/10 20:22:40 bubulle Exp $
+
+TOP=..
+CAMLP4_COMM=TOP=$(TOP) $(TOP)/tools/camlp4_comm.sh pa_ifdef.cmo
+CAMLP4F=-DUNIX
+CAMLP4D=`camlp4 -where`
+LIBUNIX=-cclib -lunix
+OCAMLC=ocamlc -warn-error A
+OCAMLOPT=ocamlopt -warn-error A
+OCAMLI=
+OCAMLD=
+STRIP=strip
+RM=/bin/rm -f
+EXE=
+STATIC=-ccopt -static
+STATIC=
+
+all::
+opt::
+
+clean::
+	$(RM) *.out *.opt *.cm[oixa] *.cmxa *.pp[oi] *.o *.a *.obj *.lib *.lck *.bak *~ .#*
+
+.SUFFIXES: .ml .cmo .mli .cmi .cmx
+
+.mli.cmi:
+	@if test `basename $<` != $<; then echo "Bad directory for $<"; exit 1; fi
+	@$(CAMLP4_COMM) $(CAMLP4F) $< -o $*.ppi
+	$(OCAMLC) $(OCAMLI) -I $(CAMLP4D) -c -intf $*.ppi
+	$(RM) $*.ppi
+
+.ml.cmo:
+	@if test `basename $<` != $<; then echo "Bad directory for $<"; exit 1; fi
+	@$(CAMLP4_COMM) $(CAMLP4F) $< -o $*.ppo
+	$(OCAMLC) $(OCAMLI) $(OCAMLD) -I $(CAMLP4D) -c -impl $*.ppo
+	$(RM) $*.ppo
+
+.ml.cmx:
+	@if test `basename $<` != $<; then echo "Bad directory for $<"; exit 1; fi
+	@$(CAMLP4_COMM) $(CAMLP4F) $< -o $*.ppo
+	$(OCAMLOPT) $(OCAMLI) -I $(CAMLP4D) -c -impl $*.ppo
+	$(RM) $*.ppo
